### PR TITLE
Aurora serverless

### PR DESCRIPTION
**Pull request by EgidioCaprino**
It adds a basic CloudFormation template for setting up a serverless Aurora cluster in a VPC and a bastion host in EC2 for accessing the database from outside the private network.

### CODE REVIEW

File: community/rds/aurora-serverless/template.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: Aurora serverless cluster
+
+Parameters:
+  DatabaseName:
+    Type: String
+  MasterUserPassword:
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: MasterUserPassword, if read from SSM Parameter Store, should be a SecureString using a dynamic reference: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html

File: community/rds/aurora-serverless/template.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Type: AWS::SSM::Parameter::Value<String>
+    Description: Parameter Store name
+  VpcId:
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: VpcId parameters should use the AWS-Specific "AWS::EC2::VPC::Id" parameter type (see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html)

File: community/rds/aurora-serverless/template.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    Type: String
+  VpcSecurityGroupId:
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: I'm having a little trouble following the networking in this example. The SG in the parameters is associated with the cluster, which is then associated with the "SecurityGroup" resource you define (which looks to be the bastion Security Group), however you end up assigning both to the bastion? Why does bastion need both groups?

Also, as with the VpcId above, this should be the AWS-Specific "AWS::EC2::SecurityGroup::Id" parameter type (see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html)

File: community/rds/aurora-serverless/template.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    Type: String
+  BastionKeyName:
+    Type: String
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: Also, as with the VpcId above, this should be the AWS-Specific "AWS::EC2::KeyPair::KeyName" parameter type (see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html)

File: community/rds/aurora-serverless/template.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      DatabaseName: !Ref DatabaseName
+      MasterUsername: root
+      MasterUserPassword: !Ref MasterUserPassword
+      DBClusterIdentifier: !Ref AWS::StackName
+      BackupRetentionPeriod: 35
+      DeletionProtection: true
+      VpcSecurityGroupIds:
+        - !Ref VpcSecurityGroupId
+
+  SecurityGroup:
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: If this is specific to the "bastion", give it a more descriptive logical name (BastionSecurityGroup?)

File: community/rds/aurora-serverless/template.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    Description: EC2 key used to connect to the bastion host
+    Default: Default
+
+Resources:
+  Cluster:
+    Type: AWS::RDS::DBCluster
+    Properties:
+      Engine: aurora
+      EngineMode: serverless
+      EngineVersion: 5.6
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: Technically, this should be a string ("5.6") -- is there any way to future-proof this? Perhaps this should be a parameter with a default as well.

File: community/rds/aurora-serverless/template.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      DatabaseName: !Ref DatabaseName
+      MasterUsername: root
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: Again, is it typical to have MasterUserName as "root" or does it make sense to parameterize?

File: community/rds/aurora-serverless/template.yml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+          FromPort: 3306
+          ToPort: 3306
+      SecurityGroupIngress: []
+      VpcId: !Ref VpcId
+
+  Bastion:
+    Type: AWS::EC2::Instance
+    Properties:
+      DisableApiTermination: true
+      ImageId: ami-035be7bafff33b6b6
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: Hard coding AMI IDs is never a good idea, plus I'm having trouble finding this particular AMI.  An easy trick is to use the managed parameter store values to grab the latest Amazon Linux AMI dynamically:
https://aws.amazon.com/blogs/compute/query-for-the-latest-amazon-linux-ami-ids-using-aws-systems-manager-parameter-store/

File: community/rds/aurora-serverless/template.yml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      InstanceType: t2.nano
+      KeyName: !Ref BastionKeyName
+      Monitoring: false
+      SecurityGroupIds:
+        - !Ref VpcSecurityGroupId
+        - !Ref SecurityGroup
+      UserData: !Base64 'yum install mysql --assumeyes'
+
+Outputs:
```
Comment (cmmeyer) [inline community/rds/aurora-serverless/template.yml]: Doesn't it mage sense to Output the endpoint/IP of the bastion?

File: community/rds/aurora-serverless/template.yml (hunk 1 above)
Comment (cmmeyer) [reply community/rds/aurora-serverless/template.yml]: FYI, this and a few of my other suggestions all come from running cfn-lint against your template:
https://pypi.org/project/cfn-lint/

Comment (EgidioCaprino) [reply community/rds/aurora-serverless/template.yml]: A linter for CF templates? Awesome, thanks :smiley:

File: community/rds/aurora-serverless/template.yml (hunk 3 above)
Comment (EgidioCaprino) [reply community/rds/aurora-serverless/template.yml]: If I remove the VPC security group from the EC2 instance I'm not able to access the database anymore.

Comment (cmmeyer) [reply community/rds/aurora-serverless/template.yml]: See, this is why I'm a CloudFormation DA, and not a networking DA. :)

Comment (EgidioCaprino) [reply community/rds/aurora-serverless/template.yml]: You're observation made sense. I tried that out but I wasn't able to connect to the database anymore. Maybe there is a better way for allowing the bastion to access the database but I honestly don't know it 😅

**Comment by EgidioCaprino** [comment]
@cmmeyer thank you for the comments and for pointing me to `cfn-lint` 😃

I've updated it. Let me know if you see something else I can improve.

**Pull request merged**